src/main/java/org/mule/extension/api/introspection/ImmutableSourceModel.java

```diff
@@ -46,13 +48,15 @@ public ImmutableSourceModel(String name,
                                 DataType attributesType,
                                 SourceFactory sourceFactory,
                                 Map<String, Object> modelProperties,
-                                List<InterceptorFactory> interceptorFactories)
+                                List<InterceptorFactory> interceptorFactories,
+                                Optional<ExceptionEnricherFactory> exceptionEnricherFactory)
```

Comment:
Can we have two constructors, one receiving the _ exceptionEnricherFactory_ and one without it? This way, when there is no factory we don't have to pass in the Optional.empty(), but instead that is done internally by the constructor with less parameters.